File: JavaScript/recorrido_insercion_busqueda.js
// 1. Recorrido
let arr = [10, 20, 30, 40, 50];
console.log("Recorrido del arreglo:");
arr.forEach(v => console.log(v));

// 2. Insertar en índice
arr.splice(2, 0, 99); // inserta 99 en la posición 2
console.log("Después de insertar:", arr);

// 3. Búsqueda lineal
let x = 30;
let encontrado = false;
for (let i = 0; i < arr.length; i++) {
    if (arr[i] === x) {
        console.log(`Elemento ${x} encontrado en índice ${i}`);
        encontrado = true;
        break;
    }
}
if (!encontrado)
    console.log(`Elemento ${x} no encontrado`);

File: Python/recorrido_insercion_busqueda.py
# 1. Recorrer e imprimir valores
arr = [10, 20, 30, 40, 50]
print("Recorrido del arreglo:")
for i in arr:
    print(i)

# 2. Insertar en índice determinado
arr.insert(2, 99)   # Inserta 99 en la posición 2
print("Después de insertar:", arr)

# 3. Búsqueda lineal
x = 30
encontrado = False
for i in range(len(arr)):
    if arr[i] == x:
        print(f"Elemento {x} encontrado en índice {i}")
        encontrado = True
        break
if not encontrado:
    print(f"Elemento {x} no encontrado")
